README.md

```diff
@@ -1,37 +1,107 @@
+### [Concurrency](https://www.stlab.cc/libraries/concurrency/)
 This library provides futures and channels, high level abstractions for implementing algorithms that eases the use of multiple CPU cores while minimizing contention. This library solves several problems of the C++11 and C++17 TS futures.
 
-# Documentation
+## Documentation
 
 The complete documentation is available on the [STLab home page](http://stlab.cc).
 
 Release changelogs are listed in [CHANGES.md](CHANGES.md).
 
-# Supported Platforms
-- All OS that have a C++14 compliant compiler
+## Tested on
 
-# Requirements
-- C++14 Compiler
--- Visual Studio 2019
--- gcc (>= 7)
--- clang (>= 6)
-- [OPTIONAL] Boost.Optional and Boost.Variant. Required when `STLAB_USE_BOOST_CPP17_SHIMS` is enabled.
-- [OPTIONAL] Boost.Test. Used when `BUILD_TESTING` is enabled.
+- Linux with GCC 11
+- Linux with Clang 14
+- MacOS with Apple-clang 13.0.1
+- Windows with Visual Studio 16
 
-# Building
+## Requirements
 
-STLab is a standard CMake project. See the [running CMake](https://cmake.org/runningcmake) tutorial for an introduction to this tool.
+- A standards-compliant C++14, C++17, or C++20 compiler
+- **Use with a compiler in C++14-compliant mode** requires Boost.Optional and Boost.Variant >= 1.74.0
+- **Building** the library requires CMake 3.23 or later
+- **Testing or developing** the library requires Boost.Test >= 1.74.0
+
+## Building
+
+STLab is a standard CMake project. See the [running CMake](https://cmake.org/runningcmake) tutorial
+for an introduction to this tool.
+
+### Preparation
+
+1. Create a build directory outside this library's source tree.  In this guide, we'll use a sibling
+   directory called `BUILD`.
+
+2. If you are using the library in C++14-compliant mode or need to run the test suite, be sure you
+   have the necessary parts of Boost >= 1.74.0 installed.  Linux distributions usually make a
+   suitable version available through standard package managers.  On macOS or Linux, you can easilly
+   install Boost using [Homebrew](https://brew.sh/).  To install Boost on Windows, you can use
+   Boost's [binary installers](https://sourceforge.net/projects/boost/files/boost-binaries/).
+
+3. Install a version of CMake >= 3.23.  If you are on Debian or Ubuntu Linux you may need to use
+   `snap` to find one that's new enough.
+
+4. If you are using MSVC, you may need to set environment variables appropriately for your target
+   architecture by invoking `VCVARSALL.BAT` with an appropriate option.
+
+### Configure
+
+Run CMake in the root directory of this project, setting `../BUILD` as your build directory.  The
+basis of your command will be
+
+```
+cmake -S . -B ../BUILD
+```
+
+but there are other options you may need to append in order to be successful.  Among them:
+
+* `-DCMAKE_CXX_STANDARD=`[`14`|`17`|`20`|`23`] to build with compliance to the given C++ standard.
+* `-DSTLAB_USE_BOOST_CPP17_SHIMS=ON` if you are building in C++14 compliant mode.
```

Comment:
```suggestion
* `-DSTLAB_USE_BOOST_CPP17_SHIMS=ON` if you would like to use Boost.Optional and Boost.Variant instead of those in the standard. This is required if your compiler doesn't support the standard versions or if CMAKE_CXX_STANDARD is set to 14.
```

Reply:
I don't think it's helpful to suggest that people building for C++17 might want to use Boost.Optional and Boost.Variant.  Doesn't that just increase the likelihood of some ODR problem down the road?